great_expectations/experimental/datasources/file_path_data_asset.py

```diff
@@ -225,34 +227,39 @@ def get_batch_list_from_batch_request(
 
         return batch_list
 
+    def _build_or_get_data_connector(self) -> DataConnector:
+        """This private method ensures that exactly one instance of "DataConnector" class is available."""
+        data_connector: DataConnector
+        try:
+            data_connector = self._data_connector
+        except AttributeError:
+            data_connector = self._build_data_connector()
+            self._data_connector = data_connector
+
+        return data_connector
+
     def test_connection(self) -> None:
         """Test the connection for the DataAsset.
 
         Raises:
             TestConnectionError: If the connection test fails.
         """
-        self._insure_single_data_connector_instance_exists()
-
+        data_connector: DataConnector = self._build_data_connector()
         if (
-            self._data_connector.get_unmatched_data_reference_count()
-            == self._data_connector.get_data_reference_count()
+            data_connector.get_unmatched_data_reference_count()
```

Comment:
I think we can simplify this by:
- adding a `test_connection` method to the data_connector, returns true or false
- making this method abstract in `FilePathDataAsset`
- Leave the subclass to call `data_connector.test_connection` and craft the error.